File: example-jpms/src/test/java/example/jpms/UpdateTest.java
package example.jpms;

import static org.junit.jupiter.api.Assertions.assertNotNull;

import example.jpms.entity.Employee;
import example.jpms.entity.JobType;
import example.jpms.repository.EmployeeRepository;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.seasar.doma.jdbc.Config;

@ExtendWith(TestEnvironment.class)
public class UpdateTest {

  private final EmployeeRepository repository;

  UpdateTest(Config config) {
    repository = new EmployeeRepository(config);
  }

  @Test
  public void testUpdate() {
    Employee employee = repository.selectById(1);
    assertNotNull(employee);
    employee.setName("hoge");
    employee.setJobType(JobType.PRESIDENT);
    repository.update(employee);
  }

  @Test
  public void testUpdateByNativeSql() {
    Employee employee = repository.selectById(1);
    assertNotNull(employee);
    employee.setName("hoge");
    employee.setJobType(JobType.PRESIDENT);
    repository.updateByNativeSql(employee);
  }
}

File: dsl-style-kotlin/src/test/kotlin/example/dsl/style/kotlin/UpdateTest.kt
package example.dsl.style.kotlin

import example.dsl.style.kotlin.entity.JobType
import example.dsl.style.kotlin.repository.EmployeeRepository
import org.junit.jupiter.api.Assertions.assertNotNull
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import org.seasar.doma.jdbc.Config

@ExtendWith(TestEnvironment::class)
class UpdateTest internal constructor(config: Config) {

    private val repository: EmployeeRepository = EmployeeRepository(config)

    @Test
    fun testUpdate() {
        val employee = repository.selectById(1)
        assertNotNull(employee)
        employee.name = "hoge"
        employee.jobType = JobType.PRESIDENT
        repository.update(employee)
    }

    @Test
    fun testUpdateByNativeSql() {
        val employee = repository.selectById(1)
        assertNotNull(employee)
        employee.name = "hoge"
        employee.jobType = JobType.PRESIDENT
        repository.updateByNativeSql(employee)
    }
}
